src/tests/ftest/pool/management_race.py

```diff
@@ -0,0 +1,134 @@
+            else:
+                completed = False
+                start = float(time.time())
+                while not completed:
+                    try:
+                        self.log.info("--(%d.4.%d.%d)--pool querying.. \n",
+                                      thread_num, pool_num, test_num)
+                        query_result = self.get_dmg_command().pool_query(pool=label)
+                        self.log.info("--(%d.5.%d.%d)--get pool list all.. \n",
+                                      thread_num, pool_num, test_num)
+                        pool_list_result = self.get_dmg_command().get_pool_list_all()
+                        completed = True
+                    except CommandFailure as error:
+                        self.log.info(
+                            "-->Pool %s deleted by another thread, "
+                            "race condition detected, retry.. %s", label, error)
+                        time.sleep(wait_time)
+                        if float(time.time()) - start > max_query_time:
+                            self.fail("#(%d.5.%d.%d)Pool query failed and retry timeout.",
+                                      thread_num, pool_num, test_num)
+
+    def test_pool_management_race(self):
+        """JIRA ID: DAOS-8463: Pool management integrity with race boundary conditions
+        Test Description:
+            Verify pool management actions create/delete/query are well behaved when multiple
+            queries while deleting and recreate.
+            Using 2 thread_managers, 1st thread_manager performs repeating pool_delete and
+            pool_recreate, 2nd thread_manager (with multiple threads) perform pool_query and
+            pool_list.
+        Use case:
+            1. Create num_pools pools.
+            2. Setup the thread manager for delete and recreate the pool with same label.
+            3. Setup multiple threads query and list pools, retry when the pool been deleted
+               by the other Thread, until timeout.
+            4. Launch all the threads with number of test loops.
+            5. Check for failure from thread_manager
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium,ib2
+        :avocado: tags=boundary_test
+        :avocado: tags=pool_mgmt_race
+        Args:
+        """
+
+        num_pools = self.params.get("num_pools", '/run/boundary_test/*')
+        test_loop = self.params.get("test_loop", '/run/boundary_test/*')
+        num_query_threads = self.params.get("num_query_threads", '/run/boundary_test/*')
+        pool_label_prefix = self.params.get("pool_label_prefix", '/run/boundary_test/*')
+        self.pool = []
+        for pool_number in range(num_pools):
+            label = pool_label_prefix + str(pool_number)
+            pool = self.get_pool()
+            pool.set_property(prop_name="label", prop_value=label)
+            pool.label.update(label)
+            self.pool.append(pool)
```

Comment:
```suggestion
            self.pool.append(self.get_pool(create=False))
            self.pool[-1].label.update(label)
            self.pool[-1].create()         
```
It's more efficient to update the label to the one you want to use, then create.
Note that if you use the default label, you don't even need to do this.

Reply:
Have to use specific pool label for race condition, please see my reply above.